qiskit_dynamics/models/generator_model.py

```diff
@@ -529,3 +420,141 @@ def evaluate_rhs(self, time: float, y: Array, in_frame_basis: Optional[bool] = F
+            if old_frame.frame_operator is not None:
+                if issparse(static_operator):
+                    if old_frame.frame_operator.ndim == 1:
+                        static_operator = diags(old_frame.frame_operator, format="csr")
+                    else:
+                        static_operator = csr_matrix(old_frame.frame_operator)
+                else:
+                    if old_frame.frame_operator.ndim == 1:
+                        static_operator = np.diag(old_frame.frame_operator)
+                    else:
+                        static_operator = old_frame.frame_operator
+        # transform into new frame basis, and add the new frame operator
+        if static_operator is not None:
+            static_operator = new_frame.generator_into_frame(
+                t=0.0,
+                operator=static_operator,
+                operator_in_frame_basis=False,
+                return_in_frame_basis=True,
+            )
+        else:
+            # "subtract" the frame operator from 0
+            if new_frame.frame_operator is not None:
+                if issparse(static_operator):
+                    static_operator = -diags(new_frame.frame_diag, format="csr")
+                else:
+                    static_operator = -np.diag(new_frame.frame_diag)
+
+        return static_operator
+
+    @classmethod
+    def transfer_operators_between_frames(
+        cls,
+        operators: Union[None, Array, List[csr_matrix]],
+        new_frame: Optional[Union[Array, RotatingFrame]] = None,
+        old_frame: Optional[Union[Array, RotatingFrame]] = None,
+    ) -> Tuple[Union[None, Array]]:
+        """Transform operators for a GeneratorModel from one frame basis into another.
+
+        Args:
+            operators: Operators of the model. If None, remains as None.
+            new_frame: New rotating frame.
+            old_frame: Old rotating frame.
+
+        Returns:
+            operators: Transformed as described above.
+        """
+        new_frame = RotatingFrame(new_frame)
+        old_frame = RotatingFrame(old_frame)
+
+        operators = to_numeric_matrix_type(operators)
+
+        # transform out of old frame basis
+        if operators is not None:
+            # For sparse case, if list, loop
+            if isinstance(operators, list):
+                operators = [old_frame.operator_out_of_frame_basis(op) for op in operators]
+            else:
+                operators = old_frame.operator_out_of_frame_basis(operators)
+
+        # transform into new frame basis
+        if operators is not None:
+            # For sparse case, if list, loop
+            if isinstance(operators, list):
+                operators = [new_frame.operator_into_frame_basis(op) for op in operators]
+            else:
+                operators = new_frame.operator_into_frame_basis(operators)
+
+        return operators
+
+    @classmethod
+    def construct_operator_collection(
```

Comment:
Does this need to be part of the public API or can it be `_construct_operator_collection`. If public does it need to be a method (do subclasses overload it?) or can it be a separate function.

Reply:
Can just make it a separate function, same as above.

Reply:
Done.